reward/staking_manager.go

```diff
@@ -103,38 +103,45 @@ func GetStakingManager() *StakingManager {
 	return stakingManager
 }
 
-// GetStakingInfo returns a corresponding stakingInfo for a blockNum.
+// GetStakingInfo returns a stakingInfo on the staking block of the given block number.
+// Note that staking block is the block on which the associated staking information is stored and used during an interval.
 func GetStakingInfo(blockNum uint64) *StakingInfo {
+	stakingBlockNumber := params.CalcStakingBlockNumber(blockNum)
+	logger.Debug("Staking information is requested", "blockNum", blockNum, "staking block number", stakingBlockNumber)
+	return GetStakingInfoOnStakingBlock(stakingBlockNumber)
+}
+
+// GetStakingInfoOnStakingBlock returns a corresponding StakingInfo for a staking block number.
+// If the given number is not on the staking block, it returns nil.
+func GetStakingInfoOnStakingBlock(stakingBlockNumber uint64) *StakingInfo {
```

Comment:
hmm the function name is little bit weird...

1. Doesn't it need to check whether the block number is on the staking interval or not?
2. What about the function name of `GetStakingInfoOnStakingInterval`? `GetStakingInfoStrict`? Let me think little bit more...

Reply:
1. As the comment is described, it returns `nil` if the number is not on staking block. It should be added if we need to handle the related error.

Reply:
But I couldn't find the check logic. Where is it checked?

Reply:
Oh, I said if we need to handle it. I think we don't need to handle it. Just returning nil is fine IMO.

Reply:
@jeongkyun-oh I couldn't find the explicit condition check below. which function call treats that?

Reply:
I have updated the shortcut.